### PR TITLE
Split `BigOperatorAtom.CreateBoxCore` into smaller methods

### PR DESCRIPTION
**Pull request by Lehonti**
I think the original method is long and chaotic. Hopefully this can help improve it, even if it's not yet perfect

### CODE REVIEW

File: src/XamlMath.Shared/Atoms/BigOperatorAtom.cs [hunk 1]
```diff
@@ -23,91 +24,130 @@ private static Box ChangeWidth(Box box, double maxWidth)
...
+    }
+
+    private readonly record struct BoxForBaseAtom(Box BaseBox, double Delta)
+    {
+        internal BoxForBaseAtom AdjustedToMaxWidth(double maxWidth) =>
+            this with { BaseBox = ChangeWidth(BaseBox, maxWidth) };
+    }
+    private BoxForBaseAtom CreateBoxForBaseAtom(TexEnvironment environment)
+    {
+        if (BaseAtom is not SymbolAtom symbolAtom || BaseAtom.Type != TexAtomType.BigOperator)
```
Comment (ForNeVeR) [inline src/XamlMath.Shared/Atoms/BigOperatorAtom.cs:126]: _(Minor)_ I'd prefer these branches to be reversed. It is kinda hard to read the inverse condition, while the original "what to do if the base atom is a symbol atom and has a type of `BigOperator`) has lower cognitive load IMO.

What do you think?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]
Thank you, it indeed became better this way! So far, I only have a minor thing to discuss.

File: src/XamlMath.Shared/Atoms/BigOperatorAtom.cs (hunk 1 above)
Comment (Lehonti) [reply src/XamlMath.Shared/Atoms/BigOperatorAtom.cs]: I must have done that as an intermediate step in some refactoring and then forgotten about it :) but yeah, I agree, inverted logic (i.e. `is not` and `!=`) may be harder to understand. I've now applied your suggestion

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**